mangadex_downloader/format/base.py

```diff
@@ -102,18 +124,19 @@ def get_images(self, chap_class, images, path, count):
                 # If file still in intact and same as the server
                 # Continue to download the others
                 if verified and not self.replace:
-                    log.debug(f"Page {page} ({img_name}) is exist and verified, cancelling download...")
+                    pbm.logger.debug(f"Page {page} ({img_name}) is exist and verified, cancelling download...")
                     count.increase()
                     imgs.append(img_path)
+                    pages_pb.update(1)
                     continue
                 elif verified == False and not self.replace:
                     # File is not same server, probably modified
-                    log.warning(
+                    pbm.logger.warning(
                         f"Page {page} ({img_name}) is exist but failed to verify (hash is not matching), " \
```

Comment:
Correct me if i'm wrong, but isn't `page` considered as singular ? why `exists` is being used instead of `is exist` ?

I'm sorry if i ask you this, i'm not perfect in english grammar.

Reply:
It's a verb, not an adjective (and `is exist` does not exist in English). The subject is "page" and the verb is `to exist`, so when you conjugate the present 3rd singular form, you get `this page exists/does not exist`.